src/DB_Command.php

```diff
@@ -495,24 +495,40 @@ public function cli( $_, $assoc_args ) {
 	 *     +---------+-----------------------+
 	 */
 	public function query( $args, $assoc_args ) {
-
 		$command = sprintf( '/usr/bin/env mysql%s --no-auto-rehash', $this->get_defaults_flag_string( $assoc_args ) );
 		WP_CLI::debug( "Running shell command: {$command}", 'db' );
-
 		$assoc_args['database'] = DB_NAME;
 
-		// The query might come from STDIN.
 		if ( ! empty( $args ) ) {
 			$assoc_args['execute'] = $args[0];
 		}
 
 		if ( isset( $assoc_args['execute'] ) ) {
-			// Ensure that the SQL mode is compatible with WPDB.
 			$assoc_args['execute'] = $this->get_sql_mode_query( $assoc_args ) . $assoc_args['execute'];
 		}
 
+		$is_update_or_delete = isset( $assoc_args['execute'] ) && preg_match( '/\b(UPDATE|DELETE|INSERT)\b/i', $assoc_args['execute'] );
```

Comment:
This can also include `REPLACE` and `LOAD DATA` and the variable name should be changed since it isn't just for update or delete queries anymore. Something more general like `$is_row_modifying_query` instead.